codelab-master/apps/playground/src/app/quiz/components/question/question.component.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+  SimpleChanges
+} from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+
+import { QuizQuestion } from '@codelab-quiz/shared/models/QuizQuestion.model.ts';
+import { QuizService, TimerService } from '@codelab-quiz/shared/services/*';
+
+@Component({
+  selector: 'codelab-quiz-question',
+  templateUrl: './question.component.html',
+  styleUrls: ['./question.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class QuizQuestionComponent implements OnInit, OnChanges {
+  currentQuestion: QuizQuestion;
+  @Output() answer = new EventEmitter<number>();
+  @Input() set question(value: QuizQuestion) { this.currentQuestion = value; }
+  formGroup: FormGroup;
+  quizStarted: boolean;
+  multipleAnswer: boolean;
+  alreadyAnswered = false;
+  isAnswered: boolean;
+  isCorrectAnswerSelected = false;
+  correctAnswers = [];
+  correctMessage = '';
+  previousUserAnswers: any[] = [];
```

Comment:
can we give it a type?

Reply:
previous user answers should probably be a string[]